### PR TITLE
LL-5398 - Platform Catalog

### PR DESCRIPTION
**Pull request by rjarasson-ledger**
Adds the Platform Catalog "Browse".

Depends on: https://github.com/LedgerHQ/ledger-live-common/pull/1266

### Type

Feature

### Context

The Platform catalog uses Live Common's CatalogProvider to display a list of apps available in LLD.
You can enable "debug" apps by using ENV variable `PLATFORM_DEBUG`

It also implements bifurcation screens removal, and analytics.

JIRA ticket: https://ledgerhq.atlassian.net/browse/LL-5398
Sub tasks: 
* https://ledgerhq.atlassian.net/browse/LL-5904
* https://ledgerhq.atlassian.net/browse/LL-5907
* https://ledgerhq.atlassian.net/browse/LL-6038
* https://ledgerhq.atlassian.net/browse/LL-5600

### TODO

* [x] remove bifurcation screens
* [x] add Platform Analytics
* [x] implement new Catalog
* [x] use translated contents
* [ ] final wording and banner illustration
* [x] COMING SOON Cards
* [ ] MAYBE - More app info modal (ℹ️ icon)
* [x] Rebase on https://github.com/LedgerHQ/ledger-live-desktop/pull/3906 to remove previous integration
* [ ] MAYBE - Add an experimental switch for displaying experimental apps (useless until catalog is remotely fetched)
* [ ] ~CANCELLED - Add in-between drawer to let user accept it is an external third-party feature, maybe TOS ?~

### Parts of the app affected / Test plan

* New "Browse" section in SideBar
* Catalog with 2 stable apps- ParaSwap & Wyre - and coming soon apps (+ debugger if PLATFORM_DEBUG)
* A CTA for a Survey (Poll)
* A CTA to access developer portal
image

### CODE REVIEW

File: src/renderer/screens/platform/App.js [hunk 1]
```diff
@@ -16,28 +18,20 @@ type Props = {
...
   const { platform } = match.params;
-  const manifest = manifests[platform];
+  const manifest = useAppManifest(platform);
 
-  const handleClose = useCallback(() => history.push(`/exchange`), [history]);
+  const handleClose = useCallback(() => history.push(`/platform`), [history]);
 
   if (!manifest) {
-    return null;
+    return "oops no manifest";
```
Comment (juan-cortes) [inline src/renderer/screens/platform/App.js:29]: To me, this is not a valid solution. If we want to show an error, then render an error.
If we want to ignore this case, the redirect to another screen, but not render a screen with oops!

File: src/renderer/screens/platform/CatalogBanner.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import React from "react";
+
+import styled from "styled-components";
+import { useTranslation } from "react-i18next";
+
+import type { ThemedComponent } from "~/renderer/styles/StyleProvider";
+
+import PlatformCatalogIllu from "~/renderer/images/platform-catalog-illu.svg";
+
+import Box from "../../components/Box";
```
Comment (juan-cortes) [inline src/renderer/screens/platform/CatalogBanner.js:11]:
```suggestion
import Box from "~/renderer/components/Box";
```

File: src/renderer/screens/platform/CatalogCTA.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+// @flow
+import React from "react";
+
+import styled from "styled-components";
+import type { ThemedComponent } from "~/renderer/styles/StyleProvider";
+
+import Box from "../../components/Box";
+import Button from "../../components/Button";
```
Comment (juan-cortes) [inline src/renderer/screens/platform/CatalogCTA.js:8]:
```suggestion
import Box from "~/renderer/components/Box";
import Button from "~/renderer/components/Button";
```

File: src/renderer/screens/platform/index.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export { default } from "./Catalog";
```
Comment (juan-cortes) [inline src/renderer/screens/platform/index.js:1]: Why aren't things inside platform, or we change the screen to be inside catalog instead of this? 🤔

**Review by juan-cortes — CHANGES_REQUESTED** [review]
Nothing major, few comments

File: src/renderer/screens/platform/App.js (hunk 1 above)
Comment (rjarasson-ledger) [reply src/renderer/screens/platform/App.js:29]: Yes you are right, this is a "debug" string that was not supposed to be committed XD

File: src/config/urls.js [hunk 5]
```diff
@@ -158,6 +158,11 @@ export const urls = {
     learnMore:
       "https://www.ledger.com/academy/benefits-of-buying-crypto-through-ledger-live?utm_source=ledger_live_desktop&utm_medium=self_referral&utm_content=exchange",
   },
+  platform: {
+    developerPage:
+      "https://developers.ledger.com?utm_source=ledger_live_desktop&utm_medium=self_referral&utm_content=catalog",
+    poll: "FIXME",
```
Comment (gre) [inline src/config/urls.js:164]: ⚠️

File: src/renderer/screens/platform/App.js (hunk 1 above)
Comment (gre) [reply src/renderer/screens/platform/App.js:29]: to be fixed.

**Review by gre — CHANGES_REQUESTED** [review]

**Review by gre — DISMISSED** [review]

**Review by gre — DISMISSED** [review]

**Review by gre — DISMISSED** [review]

**Comment by gre** [comment]
@IAmMorrow i don't know why but bot don't manage to run and provide screenshot etc.. maybe because it's a "organisatoin fork"? would we need to recreate the PR maybe?

**Pull request merged**